File: lib/src/models/Product.dart
class Product {
  final String id;
  final String name;
  final double price;
  final List images;
  final String dimensions;
  final String description;
  bool isFavorite;

  Product({
    this.id,
    this.name,
    this.price,
    this.images,
    this.dimensions,
    this.description,
    this.isFavorite,
  });
}

File: api/models/Product.js
const mongoose = require('mongoose');

const ProductSchema = new mongoose.Schema({
    name: {
        type: String,
    },
    price: {
        type: Number,
    },
    dimensions: {
        type: String,
    },
    productDescription: {
        type: String
    },
    image: {
        type: [String]
    },
    isFavorite: {
        type: Boolean
    }

});


module.exports = mongoose.model('Product', ProductSchema);